### PR TITLE
Migrate jenkins pipeline to warnings-ng, resolves #147

### LINKED ISSUE #147: update warnings plugin usage
**Is your feature request related to a problem? Please describe.**
The warnings plugin in Jenkins is deprecated. We should use the warnings-ng-plugin.
It uses the command `recordIssues` instead of `warnings`.

### PR DESCRIPTION
**Pull request by NeoTheThird**
# Description

Replaces the deprecated warnings jenkins plugin with warnings-ng.

## Resolved Issues # (issue)

- #147

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- Built on jenkins

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -26,7 +26,7 @@ pipeline{
                         stash includes: 'build-doxygen/doc_doxygen/html/**', name: 'doxydocs'
 
                         // get doxygen warnings
-                        warnings canComputeNew: false, canResolveRelativePaths: false, categoriesPattern: '', defaultEncoding: '', excludePattern: '.*README.*', healthy: '', includePattern: '', messagesPattern: '', parserConfigurations: [[parserName: 'Doxygen', pattern: 'build-doxygen/DoxygenWarningLog.txt']], unHealthy: '', failedTotalAll: '0'
+                        recordIssues filters: [excludeFile('.*README.*')], tools: [doxygen(pattern: 'build-doxygen/DoxygenWarningLog.txt')]
```
Comment (SteffenSeckler) [inline Jenkinsfile]: excludeFile should be either excludeCategory or excludeType

Comment (SteffenSeckler) [reply Jenkinsfile]: probably an `unstableTotalAll: 1` is also missing

File: Jenkinsfile [hunk 2]
```diff
@@ -183,7 +183,7 @@ pipeline{
             }
             post{
                 always{
-					warnings canComputeNew: false, categoriesPattern: '', defaultEncoding: '', excludePattern: '', healthy: '', includePattern: '', messagesPattern: '', parserConfigurations: [[parserName: 'Clang (LLVM based)', pattern: 'build*/buildlog_clang.txt'], [parserName: 'GNU Make + GNU C Compiler (gcc)', pattern: 'build*/buildlog.txt'], [parserName: 'Intel C Compiler', pattern: 'build*/buildlog_intel.txt']], unHealthy: '', unstableTotalAll: '0', unstableTotalHigh: '0', unstableTotalLow: '0', unstableTotalNormal: '0'
+                    recordIssues(tools: [clang(pattern: 'build*/buildlog_clang.txt'), gcc3(pattern: 'build*/buildlog.txt'), gcc4(pattern: 'build*/buildlog.txt'), intel(pattern: 'build*/buildlog_intel.txt')])
```
Comment (SteffenSeckler) [inline Jenkinsfile]: only gcc4, no gcc3 needed, also `unstableTotalAll: 1` is missing

File: Jenkinsfile (hunk 1 above)
Comment (SteffenSeckler) [reply Jenkinsfile]: ok, seems like the filter already works. the unstable result should probably be testet

Comment (NeoTheThird) [reply Jenkinsfile]: > probably an `unstableTotalAll: 1` is also missing

Are you sure? Before it was set to zero, which in warnings-ng is the implicit default.

Comment (SteffenSeckler) [reply Jenkinsfile]: could you just try what happens if you remove some comment, so enforce a doxygen warning?

Comment (SteffenSeckler) [reply Jenkinsfile]: push

Comment (SteffenSeckler) [inline Jenkinsfile]:
```suggestion
                        recordIssues filters: [excludeFile('.*README.*')], tools: [doxygen(pattern: 'build-doxygen/DoxygenWarningLog.txt')], unstableTotalAll: 1
```

File: Jenkinsfile (hunk 2 above)
Comment (SteffenSeckler) [inline Jenkinsfile]:
```suggestion
                    recordIssues tools: [clang(pattern: 'build*/buildlog_clang.txt'), gcc3(pattern: 'build*/buildlog.txt'), gcc4(pattern: 'build*/buildlog.txt'), intel(pattern: 'build*/buildlog_intel.txt')], unstableTotalAll: 1
```

**Pull request merged**